src/bootstrap/doc.rs

```diff
@@ -834,9 +842,10 @@ macro_rules! tool_doc {
                 cargo.arg("-Zskip-rustdoc-fingerprint");
                 // Only include compiler crates, no dependencies of those, such as `libc`.
                 cargo.arg("--no-deps");
-                cargo.arg("--lib");
+
                 $(
-                    cargo.arg("-p").arg($krate);
+                    let args: Vec<&str> = $extra_arg.split_whitespace().collect();
```

Comment:
there's no point in making this a Vec if you're going to call split whitespace - the interface should either be `["-p", "rustdoc", "-p", "rustdoc-json-types"]` or `"-p rustdoc -p rustdoc-json-types`. slight preference towards the latter but i don't feel strongly.